src/cgutils.cpp

```diff
@@ -1106,6 +1106,12 @@ static std::pair<Value*, bool> emit_isa(jl_codectx_t &ctx, const jl_cgval_t &x,
         return std::make_pair(ConstantInt::get(T_int1, *known_isa), true);
     }
 
+    if (jl_is_type_type(intersected_type) && jl_pointer_egal(intersected_type)) {
+        // Use the check in `jl_pointer_egal` to see if the type enclosed
+        // has unique pointer value.
```

Comment:
Ah, nice, so this is the stronger version of `is_uniquerep_Type` apparently!

Reply:
So I should/can update and use that instead? Seems that this doesn't handle `Type{Union{}}` correctly...